backends/ebpf/psa/externs/ebpfPsaHashAlgorithm.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2022-present Orange
+Copyright 2022-present Open Networking Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "ebpfPsaHashAlgorithm.h"
+#include "backends/ebpf/ebpfProgram.h"
+#include "backends/ebpf/ebpfType.h"
+
+namespace EBPF {
+
+EBPFHashAlgorithmPSA::argumentsList EBPFHashAlgorithmPSA::unpackArguments(
+        const IR::MethodCallExpression * expr, int dataPos) {
+    BUG_CHECK(expr->arguments->size() > ((size_t) dataPos),
+              "Data position %1% is outside of the arguments: %2%", dataPos, expr);
+
+    std::vector<const IR::Expression *> arguments;
+
+    if (expr->arguments->at(dataPos)->expression->is<IR::StructExpression>()) {
+        auto argList = expr->arguments->at(dataPos)->expression->to<IR::StructExpression>();
+        for (auto field : argList->components)
+            arguments.push_back(field->expression);
```

Comment:
@kmateuszssak we could eventually simplify that as follows:

```suggestion
    if (auto argList = expr->arguments->at(dataPos)->expression->to<IR::StructExpression>()) {
        for (auto field : argList->components)
            arguments.push_back(field->expression);
```